include/alpaka/rand/RandGenericSycl.hpp

```diff
@@ -0,0 +1,248 @@
+/* Copyright 2023 Luca Ferragina, Aurora Perego, Jan Stephan, Andrea Bocci
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+#pragma once
+
+#if defined(ALPAKA_ACC_SYCL_ENABLED)
+
+#    include <alpaka/core/BoostPredef.hpp>
+#    include <alpaka/core/Concepts.hpp>
+#    include <alpaka/dev/DevGenericSycl.hpp>
+#    include <alpaka/rand/Traits.hpp>
+
+// Backend specific imports.
+#    include <CL/sycl.hpp>
+#    if BOOST_COMP_CLANG
+#        pragma clang diagnostic push
+#        pragma clang diagnostic ignored "-Wcast-align"
+#        pragma clang diagnostic ignored "-Wcast-qual"
+#        pragma clang diagnostic ignored "-Wextra-semi"
+#        pragma clang diagnostic ignored "-Wfloat-equal"
+#        pragma clang diagnostic ignored "-Wold-style-cast"
+#        pragma clang diagnostic ignored "-Wreserved-identifier"
+#        pragma clang diagnostic ignored "-Wreserved-macro-identifier"
+#        pragma clang diagnostic ignored "-Wsign-compare"
+#        pragma clang diagnostic ignored "-Wundef"
+#    endif
+#    include <oneapi/mkl/rng.hpp>
+
+#    include <oneapi/dpl/random>
+#    if BOOST_COMP_CLANG
+#        pragma clang diagnostic pop
+#    endif
+
+#    include <type_traits>
+
+namespace alpaka::rand
+{
+    //! The SYCL rand implementation.
+    template<typename TDim>
+    class RandGenericSycl : public concepts::Implements<ConceptRand, RandGenericSycl<TDim>>
+    {
+    public:
```

Comment:
```suggestion
    struct RandGenericSycl : concepts::Implements<ConceptRand, RandGenericSycl<TDim>>
    {
```

If you define a class, but start with `public:` afterwards, you can just as well use `struct` and save a LOC :) Applies to other places in this PR as well.

Reply:
For this and rest of the file I used as a reference `RandUniformCudaHipRand.hpp` that defines it as a class (I don't know if there was a particular reason to do it like that). Anyway, I'll use the `struct` and look also at the other places where this could be applied.